gtsfm/utils/geometry_comparisons.py

```diff
@@ -162,15 +163,24 @@ def compare_global_poses(
     #  We set frame "a" the target/reference
     aTi_list_ = align_poses_sim3(aTi_list, bTi_list)
 
-    return all(
-        [
-            (
-                aTi.rotation().equals(aTi_.rotation(), rot_err_thresh)
-                and np.allclose(aTi.translation(), aTi_.translation(), rtol=trans_err_thresh,)
-            )
-            for (aTi, aTi_) in zip(aTi_list, aTi_list_)
-        ]
-    )
+    rotations_equal = all([
+        aTi.rotation().equals(aTi_.rotation(), rot_err_thresh) for (aTi, aTi_) in zip(aTi_list, aTi_list_)
+    ])
+    translations_equal = all([
+        np.allclose(aTi.translation(), aTi_.translation(), atol=trans_err_thresh) for (aTi, aTi_) in zip(aTi_list, aTi_list_)
+    ])
+    if verbose:
+        # GTSAM Rot3.equals() compares rotation matrices by absolute tolerance
+        rotation_errors = np.array([
+            (aTi.rotation().matrix() - aTi_.rotation().matrix()).max() for (aTi, aTi_) in zip(aTi_list, aTi_list_)
+        ])
+        translation_errors = np.array([
+            np.linalg.norm(aTi.translation() - aTi_.translation()) for (aTi, aTi_) in zip(aTi_list, aTi_list_)
+        ])
```

Comment:
i think we should move to angular errors? Or L2 errors in quaternions/eular angles? 

Comparing matrices for rotations do not make much sense.

Reply:
I agree. But I haven't changed that code for now. How about I open an issue, and we address it later?

Reply:
Makes sense. Thanks!

Reply:
I opened an issue